src/services/textChanges.ts

```diff
@@ -241,6 +249,13 @@ namespace ts.textChanges {
             return this;
         }
 
+        public replaceNodes(sourceFile: SourceFile, oldNodes: Node | Node[], newNodes: Node | Node[], options: ChangeNodeOptions & ChangeNodesOptions = {}) {
+            const startPosition = getAdjustedStartPosition(sourceFile, isArray(oldNodes) ? oldNodes[0] : oldNodes, options, Position.Start);
+            const endPosition = getAdjustedEndPosition(sourceFile, isArray(oldNodes) ? lastOrUndefined(oldNodes) : oldNodes, options);

```

Comment:
`lastOrUndefined(ar)` returns `undefined` if either the `ar` is undefined or `ar` is empty. We also have a `firstOrUndefined(ar)` that does the same thing except it reads the first element.

Generally, `lastOrUndefined` is just a shortcut for `ar[ar.length - 1]`.